WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutPickerViewModel.kt

```diff
@@ -0,0 +1,261 @@
+package org.wordpress.android.ui.layoutpicker
+
+import android.os.Bundle
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.wordpress.android.ui.PreviewMode
+import org.wordpress.android.ui.PreviewMode.MOBILE
+import org.wordpress.android.ui.PreviewMode.TABLET
+import org.wordpress.android.ui.PreviewMode.valueOf
+import org.wordpress.android.ui.PreviewModeHandler
+import org.wordpress.android.ui.layoutpicker.LayoutPickerUiState.Content
+import org.wordpress.android.viewmodel.Event
+import org.wordpress.android.viewmodel.ScopedViewModel
+import org.wordpress.android.viewmodel.SingleLiveEvent
+
+private const val FETCHED_LAYOUTS = "FETCHED_LAYOUTS"
+private const val FETCHED_CATEGORIES = "FETCHED_CATEGORIES"
+private const val SELECTED_LAYOUT = "SELECTED_LAYOUT"
+private const val SELECTED_CATEGORIES = "SELECTED_CATEGORIES"
+private const val PREVIEW_MODE = "PREVIEW_MODE"
+
+abstract class LayoutPickerViewModel(
+    open val mainDispatcher: CoroutineDispatcher,
+    open val bgDispatcher: CoroutineDispatcher
+) : ScopedViewModel(bgDispatcher),
+        PreviewModeHandler {
+    lateinit var layouts: List<LayoutModel>
+    lateinit var categories: List<LayoutCategoryModel>
+
+    private val _uiState: MutableLiveData<LayoutPickerUiState> = MutableLiveData()
+    val uiState: LiveData<LayoutPickerUiState> = _uiState
+
+    private val _previewMode = SingleLiveEvent<PreviewMode>()
+    val previewMode: LiveData<PreviewMode> = _previewMode
+
+    private val _onCategorySelected = MutableLiveData<Event<Unit>>()
+    val onCategorySelected: LiveData<Event<Unit>> = _onCategorySelected
+
+    private val _onThumbnailModeButtonPressed = SingleLiveEvent<Unit>()
+    val onThumbnailModeButtonPressed: LiveData<Unit> = _onThumbnailModeButtonPressed
+
+    val isLoading: Boolean
+        get() = _uiState.value === LayoutPickerUiState.Loading
+
+    abstract fun fetchLayouts()
+
+    open fun trackPreviewModeChanged(mode: String) {
+        // Tracked in subclass
+    }
+
+    open fun trackThumbnailModeTapped(mode: String) {
+        // Tracked in subclass
+    }
```

Comment:
Yes, I've left this as a default implementation to avoid having to add empty methods in `ModalLayoutPickerViewModel`. I will convert those along with the [analytics enhancements](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3130). wdyt?